[Jweewee] iP

Jweewee opened the pull request:
### **Wiz**

> "Your mind is for having ideas, not holding them." - David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

Wiz frees your mind from having to remember things you need to do. It's:

- Text-based
- Easy to learn
- ~FAST~ _SUPER_ FAST to use

All you need to do is:

1. Download it from [here](#download-it-from-here).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X] - Managing tasks
- [ ] - Managing deadlines (coming soon)
- [ ] - Reminders (coming soon)

If you are a Java programmer, you can use it ot practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Jweewee committed:
Level 0

Jweewee committed:
Level 1

Jweewee committed:
Level 2

Jweewee committed:
Level 2

Jweewee committed:
Level 3

Jweewee committed:
Level 4

Jweewee committed:
A-TextUiTesting

Jweewee committed:
Level 5

Jweewee committed:
Level 6

Jweewee committed:
Doc

Jweewee committed:
Corrected code for delete function

Jweewee committed:
branch-level-7 commit

Jweewee committed:
Merge branch-Level-7 to local Master

Jweewee committed:
branch-level-7 commit

Jweewee committed:
Merge branch 'branch-Level-7' no 2

TehOPanas commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,18 @@
+public class Deadline extends Task{
+    protected String by;
```
I feel like a more intuitive name can be given to **by**. 
For example, **endDate** or **deadline**.

TehOPanas commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,245 @@
+import java.io.*;
```
I think it will be clearer if you explicitly import the classes you use instead of the whole package, so we can get a grasp of what you're using from the start.

TehOPanas commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,245 @@
+     * @param command The user's command to be processed.
+     * @throws DukeException If the command is not recognised, or error
+     */
+    public void executeCommand(String command, String filePath) throws DukeException {
```
For each if else case, I feel like you can add a bit more spacing for higher readability.

TehOPanas commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,245 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Duke wizzer = new Duke();
```
Same thing, I prefer more spacing.

TehOPanas reviewed (commented):
Overall did a great job on all the naming conventions and coding standard, but can use a bit more spacing in between codes. **GREAT JOB!!!!**

Jweewee committed:
branch-level-8 commit

Jweewee committed:
Merge branch 'branch-Level-8'

jean-cq commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,228 @@
+            for (int i = 0; i < tasks.size(); i++) {
+                System.out.println(" " + (i + 1) + ". " + tasks.get(i).toString());
+            }
+        } else if (command.startsWith("mark") || command.startsWith("unmark")) {
```
It could be better if you put the condition as a field/variable, according to the "Code quality" under "Implementation" chapter of the textbook.
>Avoid complicated expressions

jean-cq commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,228 @@
+        }
+        System.out.println("Hello from " + logo +
+                "\nWhat can I do for you?\n");
+        System.out.println("--------------------------");
```
Since you have quite a lot of dashed lines to print, will it be better to create a field for them to avoid complicating the code?

jean-cq commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,228 @@
+            myReader.close();
+            return tasks;
+        } catch (FileNotFoundException e) {
+            createDataFolderIfNeeded(folderPath);
+            createDataFileIfNeeded(filePath);
+            writeLine(filePath);
```
I like the way you use the same level of abstraction here.

jean-cq commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,38 @@
+ * It contains a description and a time interval during which the event occurs.
+ */
+public class Event extends Task {
+    protected String start;
```
A better naming of the field would be dateStart?

jean-cq commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,24 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
+    public Task(String description) {
```
For statements of different types, will it be better to leave a blank line in between?

jean-cq commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,228 @@
+                    if (description.length() == 0) {
+                        throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+                    }
+                    String byID = parts[1].trim();
```
The naming of byID seems a bit confusing to the reader, would it be better to change it to dueDate?

jean-cq commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,228 @@
+                        + "\n" + " Now you have " + tasks.size() + " tasks in the list.");
+            }
+            writeLine(filePath);
+        } else {
```
these if-else statements can be greatly improved to a clearer presentation after you do A-MoreOOP. Do not nest too many things inside a method.

jean-cq reviewed (commented):
LGTM, efforts are seen in doing abstraction, but still can do more, especially after implementing OOP. Namings can be improved.

Jweewee committed:
A-MoreOOP merge commit

Jweewee committed:
Merge branch 'A-MoreOOP'

Jweewee committed:
A-Packages

Jweewee committed:
Merge branch 'branch-A-Packages'

Jweewee committed:
Add A-Gradle

Jweewee committed:
Merge branch 'add-gradle-support'

Jweewee committed:
branch-A-Gradle

Jweewee committed:
Merge branch 'branch-A-Gradle'

Jweewee committed:
branch-A-JUnit

Jweewee committed:
Merge branch 'branch-A-JUnit'

Jweewee committed:
branch-A-Jar

Jweewee committed:
Merge branch 'branch-A-Jar'

Jweewee committed:
branch-A-JavaDoc

Jweewee committed:
branch-Level-9

Jweewee committed:
Merge branch 'branch-Level-9'

Jweewee committed:
branch-A-JavaDoc

Jweewee committed:
branch-A-CodingStandard

Jweewee committed:
Merge branch 'branch-A-CodingStandard'

Jweewee committed:
branch-A-CodingStandard

Jweewee committed:
Merge branch 'branch-A-CodingStandard'

Jweewee committed:
branch-Level-10

Jweewee committed:
branch-Level-10

Jweewee committed:
Merge branch 'branch-Level-10'

Jweewee committed:
Merge remote-tracking branch 'origin/branch-Level-10' into branch-Level-10

Jweewee committed:
Merge branch 'branch-Level-10'

Jweewee committed:
branch-A-CodeQuality

Jweewee committed:
Added assert statement for improved debugging

In this commit, I  introduced an assert statement to enhance the reliability and debugging capabilities of the Java program. This assert statement checks a critical condition at a specific point in the code, ensuring that our assumptions about the program's state hold true during runtime.
@jweewee

Jweewee committed:
Merge pull request #3 from Jweewee/branch-A-CodeQuality

A-CodeQuality

Jweewee committed:
Merge branch 'master' into branch-A-Assertions

Jweewee committed:
Merge branch 'branch-A-Assertions'

Jweewee committed:
Added friendlier syntax for commands.

In this commit, I introduced command syntax shortcuts that are shorter aliases for keywords e.g., t can be shorter for todo.

Having friendlier syntax improved quality of life and convenience for the user.

Jweewee committed:
Merge branch 'branch-C-FriendlierSyntax'

Jweewee committed:
Added unique personality for the chatbot.

In this commit, I introduced a name for Duke and tweaked the phrases used to resemble Ms Frizzle from the Magic School Bus.

Jweewee committed:
Merge branch 'branch-A-Personality'

Jweewee committed:
Added a User Guide for this chatbot.

In this commit, I introduced a user guide for the chatbot that provides the user with enough guidance to be able to use all important features of the chatbot.

Jweewee committed:
Merge branch 'branch-A-UserGuide'
